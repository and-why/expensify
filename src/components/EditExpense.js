import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";
import RemoveModal from "./RemoveModal";

export class EditExpense extends React.Component {
  state = {
    modalOpen: false
  };
  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };
  onRemove = expense => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push("/");
  };

  handelOpenModal = () => {
    this.setState({ modalOpen: true });
  };
  handelCloseModal = () => {
    this.setState({ modalOpen: false });
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
          <button className="btn btn--red" onClick={this.handelOpenModal}>
            Remove Expense
          </button>
          <RemoveModal
            modalOpen={this.state.modalOpen}
            handelCloseModal={this.handelCloseModal}
            onRemove={this.onRemove}
          />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id == props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: data => dispatch(startRemoveExpense(data))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpense);
