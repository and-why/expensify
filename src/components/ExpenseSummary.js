import React from 'react';
import { connect } from 'react-redux'; 
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
  <div className="page-header">
    <div className="content-container">
      {
        (props.expenseCount > 1 || props.expenseCount === 0) ? 
        (
          <h1 className="page-header__title">Viewing {props.expenseCount} expenses totalling {numeral(props.expensesTotal/100).format('$0,0.00')}</h1>
        ) : (
          <h1 className="page-header__title">Viewing 1 expense totalling {numeral(props.expensesTotal/100).format('$0,0.00')}</h1>
        )
      }
    </div> 
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpenseSummary);


