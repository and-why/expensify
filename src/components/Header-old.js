import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';
import { firebase } from "../firebase/firebase";
    
let userImageURL = null;
    
firebase.auth().onAuthStateChanged(user => {
    if (user) {
        return userImageURL = user.photoURL;
    } 
});

export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Expensify</h1>
								</Link>
									<button className="header__profile btn btn--clear" onClick={startLogout}><div className="header__profile--photo"><img src={userImageURL} alt=""/></div> Logout</button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout()),
});




export default connect(undefined, mapDispatchToProps)(Header);