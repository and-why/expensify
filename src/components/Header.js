import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";
import { firebase } from "../firebase/firebase";
import CustomModal from "../components/customModal";

let userImageURL = null;

firebase.auth().onAuthStateChanged(user => {
  if (user) {
    return (userImageURL = user.photoURL);
  }
});

export class Header extends React.Component {
  state = {
    modalOpen: false
  };
  handelOpenModal = () => {
    this.setState({ modalOpen: true });
  };
  handelCloseModal = () => {
    this.setState({ modalOpen: false });
  };
  onLogout = () => {
    this.props.startLogout();
  };
  render() {
    return (
      <header className="header">
        <div className="content-container">
          <div className="header__content">
            <Link className="header__title" to="/dashboard">
              <h1>Expense Tracker</h1>
            </Link>
            <button
              className="header__profile btn btn--clear"
              onClick={this.handelOpenModal}
            >
              <div className="header__profile--photo">
                <img src={userImageURL} alt="" />
              </div>{" "}
              Logout
            </button>
            <CustomModal
              modalOpen={this.state.modalOpen}
              handelCloseModal={this.handelCloseModal}
              modalAction={this.onLogout}
            />
          </div>
        </div>
      </header>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(Header);
