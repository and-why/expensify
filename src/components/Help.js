import React from 'react';

const HelpPage = () => (
    <div>
        This is from add help component
    </div>
);

export default HelpPage;