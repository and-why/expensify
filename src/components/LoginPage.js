import React from "react";
import { connect } from "react-redux";
import { startLogin, startLoginFB } from "../actions/auth";

export const LoginPage = ({ startLogin, startLoginFB }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expense Tracker</h1>
      <p>Get your expenses in order ready for Tax Time.</p>
      <button className="btn" onClick={startLogin}>
        Login with Google
      </button>
      <button className="btn" onClick={startLoginFB}>
        Login with Facebook
      </button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin()),
  startLoginFB: () => dispatch(startLoginFB())
});
export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
