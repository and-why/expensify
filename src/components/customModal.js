import React from "react";
import Modal from "react-modal";

const CustomModal = props => (
  <Modal
    isOpen={props.modalOpen}
    onRequestClose={props.handelCloseModal}
    contentLabel="Remove Expense"
    closeTimeoutMS={200}
    className="remove--modal"
    ariaHideApp={false}
    modalAction={props.modalAction}
  >
    <h3>Are You Sure?</h3>
    <button className="btn btn--red" onClick={props.modalAction}>
      Yes
    </button>
    <button className="btn btn--light-blue" onClick={props.handelCloseModal}>
      No
    </button>
  </Modal>
);

export default CustomModal;
