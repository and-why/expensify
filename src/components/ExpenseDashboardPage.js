import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpensesListFilters';
import ExpenseSummary from './ExpenseSummary';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;