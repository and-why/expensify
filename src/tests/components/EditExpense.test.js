import React from "react";
import { shallow } from "enzyme";
import { EditExpense } from "../../components/EditExpense";
import expenses from "../fixtures/expenses";
import Modal from "react-modal";

let startEditExpense, startRemoveExpense, history, wrapper;

beforeEach(() => {
  startEditExpense = jest.fn();
  startRemoveExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <EditExpense
      startEditExpense={startEditExpense}
      startRemoveExpense={startRemoveExpense}
      history={history}
      expense={expenses[2]}
    />
  );
});

test("should render EditExpense correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("Should handle startEditExpense", () => {
  wrapper.find("ExpenseForm").prop("onSubmit")(expenses[2]);
  expect(history.push).toHaveBeenLastCalledWith("/");
  expect(startEditExpense).toHaveBeenLastCalledWith(
    expenses[2].id,
    expenses[2]
  );
});

test("Should open modal when remove expense is clicked", () => {
  wrapper.find("button").simulate("click");
  expect(wrapper.find(RemoveModal).prop("isOpen")).toEqual(true);
});

test("Should open modal when remove expense is clicked", () => {
  wrapper
    .find("button")
    .at(1)
    .simulate("click");
  expect(history.push).toHaveBeenLastCalledWith("/");
  expect(startRemoveExpense).toHaveBeenLastCalledWith({
    id: expenses[2].id
  });
});
