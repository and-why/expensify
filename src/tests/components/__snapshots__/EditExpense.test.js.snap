// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpense correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Edit Expense
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <ExpenseForm
      expense={
        Object {
          "amount": 65800,
          "createdAt": 345600000,
          "description": "Car Finance",
          "id": "3",
          "note": "Mercedes a180",
        }
      }
      onSubmit={[Function]}
    />
    <button
      className="btn btn--red"
      onClick={[Function]}
    >
      Remove Expense
    </button>
    <RemoveModal
      handelCloseModal={[Function]}
      modalOpen={false}
      onRemove={[Function]}
    />
  </div>
</div>
`;
